python/pyspark/sql/connect/dataframe.py

```diff
@@ -595,6 +595,47 @@ def join(
 
     join.__doc__ = PySparkDataFrame.join.__doc__
 
+    def _joinAsOf(
```

Comment:
I have a dumb question: why this method name start with `_`?

Reply:
Because so far this is supposed to be called from pandas API's `merge_asof` only.